pandas/core/dtypes/common.py

```diff
@@ -128,10 +128,12 @@ def ensure_python_int(value: int | np.integer) -> int:
     ------
     TypeError: if the value isn't an int or can't be converted to one.
     """
-    if not is_scalar(value):
-        raise TypeError(
-            f"Value needs to be a scalar value, was type {type(value).__name__}"
-        )
+    if not (is_integer(value) or is_float(value)):
+        if not is_scalar(value):
+            raise TypeError(
+                f"Value needs to be a scalar value, was type {type(value).__name__}"
+            )
+        raise TypeError(f"Wrong type {type(value)} for value {value}")
```

Comment:
Is this new error message tested anywhere?

Reply:
yes, in RangeIndex arithmetic